parquet-column/src/main/java/parquet/column/impl/ColumnWriterV2.java

```diff
@@ -58,13 +58,14 @@ public ColumnWriterV2(
       ColumnDescriptor path,
       PageWriter pageWriter,
       int initialSizePerCol,
-      ParquetProperties parquetProps) {
+      ParquetProperties parquetProps,
+      int pageSize) {
     this.path = path;
     this.pageWriter = pageWriter;
     resetStatistics();
-    this.repetitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxRepetitionLevel()), initialSizePerCol);
-    this.definitionLevelColumn = new RunLengthBitPackingHybridEncoder(getWidthFromMaxInt(path.getMaxDefinitionLevel()), initialSizePerCol);
-    this.dataColumn = parquetProps.getValuesWriter(path, initialSizePerCol);
```

Comment:
We should tweak the initialSize here.
levels should get a tiny initial size (100 bytes?) in case they are always null or always defined.